core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java

```diff
@@ -2480,6 +2484,32 @@ private static class RegexpReplaceImplementor extends AbstractRexCallImplementor
     }
   }
 
+  /** Implementor for the {@code TO_TIMESTAMP_LTZ} function. */
```

Comment:
is an implementor necessary? it would be better if calls to TO_TIMESTAMP_LTZ were translated to existing functions. Then we would not need new methods in `SqlFunctions`

Reply:
The main reason I added the implementor was to avoid the "error proneness" you mentioned in your above comment. It checks the type of the first argument (i.e. `DATE`, `TIMESTAMP`, etc.) to ensure it does not call the wrong method via reflection.

Before I added the implementor, when I was trying to use `TO_TIMESTAMP_LTZ(numeric)`, it would end up using the method for `DATE` arguments since dates are represented as integers. This way, that confusion is avoided.